src/components/TagCloud/TagCloud.js

```diff
@@ -0,0 +1,26 @@
+import React, { PropTypes } from 'react'
+import {Tag} from '../Tag'
+
+import {styles} from './styles'
+
+const TagCloud = ({names, onClickAction}) => (
+  <div>
+    {
+      names.map((name, i) => (
+       <div {...styles.item}>
+         {onClickAction
+           ? <Tag key={i} name={name} onClickAction={(name) => onClickAction(name)} />
```

Comment:
the onclick action has to be supplied correctly as well, we need to create the function so it removes based on name - that way we only have to deal with a one dimentional array, and not rely on a specifically formatted array of objects with certain keys